app-libs/exchange-oracle/src/coingecko.rs

```diff
@@ -96,33 +122,87 @@ mod tests {
 	use super::*;
 	use core::assert_matches::assert_matches;
 
+	fn get_coingecko_coin_id(crypto_currency: &str) -> Result<String, Error> {
+		let url = CoinGeckoClient::base_url().unwrap();
+		let mut coingecko_client = CoinGeckoClient::new(url);
+		let trading_pair = TradingPair {
+			crypto_currency: crypto_currency.to_string(),
+			fiat_currency: "USD".to_string(),
+		};
+		coingecko_client.crypto_currency_id(trading_pair)
+	}
+
+	#[test]
+	fn crypto_currency_id_works_for_dot() {
+		let coin_id = get_coingecko_coin_id("DOT").unwrap();
+		assert_eq!(&coin_id, "polkadot");
+	}
+
+	#[test]
+	fn crypto_currency_id_works_for_teer() {
+		let coin_id = get_coingecko_coin_id("TEER").unwrap();
+		assert_eq!(&coin_id, "integritee");
+	}
+
+	#[test]
+	fn crypto_currency_id_works_for_ksm() {
+		let coin_id = get_coingecko_coin_id("KSM").unwrap();
+		assert_eq!(&coin_id, "kusama");
+	}
+
+	#[test]
+	fn crypto_currency_id_works_for_btc() {
+		let coin_id = get_coingecko_coin_id("BTC").unwrap();
+		assert_eq!(&coin_id, "bitcoin");
+	}
+
+	#[test]
+	fn crypto_currency_id_error_for_undefined_crypto_currency() {
```

Comment:
Rename suggestion: `crypto_currency_id_errors_for_undefined_crypto_currency` (with additional s in `error`)

Reply:
Why a s? There is one error. 
Better name would be: invalid_crypto_currency_id_error_for_undefined_crypto_currency ?

Reply:
I understood this function name as a sentence, i.e. "error" as a verb, hence the function "errors" in case of an undefined crypto currency. Just like the function names with "works" above (`crypto_currency_id_works_for_btc()` etc).

Reply:
Rename to `crypto_currency_id_fails_for_undefined_crypto_currency`